src/xibo-interactive-control.js

```diff
@@ -493,8 +519,75 @@ window.xiboIC = (function() {
         return _lib[widgetId][name].apply(null, args);
       }
     },
+
+    // Realtime data
+    /**
+     * Get realtime data from the player. Called from the widget.
+     * @param {string} dataSetId The id of the dataset
+     * @param {Object[]} [options] - Request options
+     * @param {callback} [options.done]
+     * @param {callback} [options.error]
+     */
+    getData(dataSetId, {done, error} = {}){
+      _lib.makeRequest(
+        '/realtime?dataSetId='+dataSetId,
+        {
+          type: 'GET',
+          done: done,
+          error: error,
+        },
+      );
+    },
+
+    /**
+     * Set the realtime into the player. Called from Data Connector.
+     * @param {string} dataSetId The id of the dataset
+     * @param {Object[]} data The data for the dataset as JSON array of objects
+     * @param {Object} options - Request options
+     * @param {callback} options.done Optional
+     * @param {callback} options.error Optional
+     */
+    setData(dataSetId, data, {done, error} = {}){
+      _lib.makeRequest(
+        '/realtime?dataSetId='+dataSetId,
+        {
+          type: 'POST',
+          data: data,
+          done: done,
+          error: error,
+        },
+      );
+    },
+
+    /**
+     * Notify main application that we have new data. Called from
```

Comment:
Called from the Data Connector?

Reply:
Yes. 
The data connector should provide the function 'setData' to the JavaScript code that the user enter in order to save the data somewhere.

In SoC player case, that would be a memory location in the NodeJS part of the application.

Reply:
The code comment leaves some mystery, but yes, that makes sense.